Mathlib/Algebra/GCDMonoid/Basic.lean

```diff
@@ -0,0 +1,1438 @@
+  normalize_zero
+#align associates.out_top Associates.out_top
+
+-- Porting note: linter wanted the lhs in this simp-normal form instead of `normalize a.out`
+@[simp]
+theorem normalize_out (a : Associates α) :
+  Associates.out a * ↑(normUnit (Associates.out a)) = a.out :=
+  Quotient.inductionOn a normalize_idem
+#align associates.normalize_out Associates.normalize_out
+
+@[simp]
+theorem mk_out (a : Associates α) : Associates.mk a.out = a :=
+  Quotient.inductionOn a mk_normalize
+#align associates.mk_out Associates.mk_out
+
+theorem out_injective : Function.Injective (Associates.out : _ → α) :=
+  Function.LeftInverse.injective mk_out
+#align associates.out_injective Associates.out_injective
+
+end Associates
+
+/-- GCD monoid: a `CancelCommMonoidWithZero` with `gcd` (greatest common divisor) and
+`lcm` (least common multiple) operations, determined up to a unit. The type class focuses on `gcd`
+and we derive the corresponding `lcm` facts from `gcd`.
+-/
+-- Porting note: tactic not ported
+-- @[protect_proj]
+class GCDMonoid (α : Type _) [CancelCommMonoidWithZero α] where
+  /-- The greatest common divisor between two elements. -/
+  gcd : α → α → α
+  /-- The least common multiple between two elements. -/
+  lcm : α → α → α
+  /-- The GCD is a divisor of the first element. -/
+  gcd_dvd_left : ∀ a b, gcd a b ∣ a
+  /-- The GCD is a divisor of the second element. -/
+  gcd_dvd_right : ∀ a b, gcd a b ∣ b
+  /-- Tny common divisor of both elements is a divisor of the GCD. -/
+  dvd_gcd : ∀ {a b c}, a ∣ c → a ∣ b → a ∣ gcd c b
+  /-- The product of two elements is `Associated` with the product of their GCD and LCM. -/
+  gcd_mul_lcm : ∀ a b, Associated (gcd a b * lcm a b) (a * b)
+  /-- `0` is left-absorbing. -/
+  lcm_zero_left : ∀ a, lcm 0 a = 0
+  /-- `0` is right-absorbing. -/
+  lcm_zero_right : ∀ a, lcm a 0 = 0
+#align gcd_monoid GCDMonoid
+
+/-- Normalized GCD monoid: a `CancelCommMonoidWithZero` with normalization and `gcd`
+(greatest common divisor) and `lcm` (least common multiple) operations. In this setting `gcd` and
+`lcm` form a bounded lattice on the associated elements where `gcd` is the infimum, `lcm` is the
+supremum, `1` is bottom, and `0` is top. The type class focuses on `gcd` and we derive the
+corresponding `lcm` facts from `gcd`.
+-/
+class NormalizedGCDMonoid (α : Type _) [CancelCommMonoidWithZero α] extends NormalizationMonoid α,
+  GCDMonoid α where
+  /-- The GCD is normalized to itself. -/
+  normalize_gcd : ∀ a b, normalize (gcd a b) = gcd a b
+  /-- The LCM is normalized to itself. -/
+  normalize_lcm : ∀ a b, normalize (lcm a b) = lcm a b
+#align normalized_gcd_monoid NormalizedGCDMonoid
+
+export GCDMonoid (gcd lcm gcd_dvd_left gcd_dvd_right dvd_gcd lcm_zero_left lcm_zero_right)
+
+attribute [simp] lcm_zero_left lcm_zero_right
+
+section GCDMonoid
+
+variable [CancelCommMonoidWithZero α]
+
+-- Porting note: linter wanted the lhs in this simp-normal form instead of `normalize (gcd a b)`
```

Comment:
Hmm, I'm not sure that's the best solution

Reply:
I made a thread on Zulip about this: https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/simp.20normal.20form/near/318018031

Reply:
I ended up keeping the original lemma statements and using a lower `simp` priority on them. Hope this is resolved.